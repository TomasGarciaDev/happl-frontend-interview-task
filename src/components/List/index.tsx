import React, { useEffect, useState } from 'react'
import Card from '@/components/Card'
import { Allowance } from '@/types/allowances'

const List = () => {
  const [allowances, setAllowances] = useState<Allowance[]>([])

  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch('/api/allowances')
      const { result } = await response.json()
      setAllowances(result)
    }

    fetchData()
  }, [])

  return (
    <div className="p-8 w-full">
      <h2 className="font-semibold text-2xl leading-5">Allowances</h2>

      <div className="grid grid-cols-3 py-8 gap-4">
        {allowances.map((allowance) => (
          <Card key={allowance.id} allowance={allowance}></Card>
        ))}
      </div>

      <div className="py-8">
        {allowances.map((allowance) => (
          <div key={allowance.id}>{JSON.stringify(allowance)}</div>
        ))}
      </div>
    </div>
  )
}

export default List
