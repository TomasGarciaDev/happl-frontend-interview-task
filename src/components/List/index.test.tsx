import { render, waitFor, screen } from '@testing-library/react'
import List from '@/components/List'
import '@testing-library/jest-dom'
import 'jest-fetch-mock'
import data from '@/data/data.json'

describe('List', () => {
  beforeEach(() => {
    fetchMock.mockResponse(JSON.stringify({ result: data }), { status: 200 })
  })

  it('renders as expected', async () => {
    const { container } = render(<List />)
    await waitFor(() => expect(container).toMatchSnapshot())
  })

  it('renders the Allowances heading', async () => {
    render(<List />)

    const heading = await screen.findByRole('heading', { name: /Allowances/i })
    expect(heading).toBeInTheDocument()
  })
})
