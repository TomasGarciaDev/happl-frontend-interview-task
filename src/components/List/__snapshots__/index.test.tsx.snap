// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<div>
  <div
    class="p-8 w-full"
  >
    <h2
      class="font-semibold text-2xl leading-5"
    >
      Allowances
    </h2>
    <div
      class="grid grid-cols-3 py-8 gap-4"
    />
    <div
      class="py-8"
    />
  </div>
</div>
`;
